rust/hogvm/src/utils.rs

```diff
@@ -0,0 +1,34 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum HogVMException {
+    #[error("HogVMException: {0}")]
+    Exception(String),
+}
+
+#[derive(Debug, Error)]
+pub enum UncaughtHogVMException {
+    #[error("UncaughtHogVMException: {0}")]
+    Exception(String),
+}
+
+pub fn calculate_cost(object: &serde_json::Value) -> u64 {
+    match object {
+        serde_json::Value::Null => 1,
+        serde_json::Value::Bool(_) => 1,
+        serde_json::Value::Number(_) => 1,
+        serde_json::Value::String(s) => s.len() as u64,
+        serde_json::Value::Array(arr) => arr.iter().map(|v| calculate_cost(v)).sum(),
+        serde_json::Value::Object(obj) => obj.iter().map(|(k, v)| k.len() as u64 + calculate_cost(v)).sum(),
```

Comment:
This might make sense? I dont know how closely you're trying to match the existing HogVM code vs. just building it "the crab way"

```suggestion
        serde_json::Value::String(s) => s.len() as u64,
        serde_json::Value::Array(arr) => arr.iter().map(|v| calculate_cost(v)).sum(),
        serde_json::Value::Object(obj) => obj.iter().map(|(k, v)| k.len() as u64 + calculate_cost(v)).sum(),
        _ => 1
```